$OpenBSD: patch-src_locked_rs,v 1.1.1.1 2021/01/30 19:16:45 abieber Exp $

Disable region locking on OpenBSD, it's not supported by region-rs and not
really needed.

--- src/locked.rs.orig	Wed Dec  2 10:30:16 2020
+++ src/locked.rs	Fri Jan 22 17:19:15 2021
@@ -1,10 +1,12 @@
 use zeroize::Zeroize;
 
+#[cfg(not(target_os = "openbsd"))]
 pub struct Vec {
     data: Box<arrayvec::ArrayVec<[u8; 4096]>>,
     _lock: region::LockGuard,
 }
 
+#[cfg(not(target_os = "openbsd"))]
 impl Default for Vec {
     fn default() -> Self {
         let data = Box::new(arrayvec::ArrayVec::<_>::new());
@@ -13,6 +15,12 @@ impl Default for Vec {
         let lock = region::lock(data.as_ptr(), data.capacity()).unwrap();
         Self { data, _lock: lock }
     }
+}
+
+#[cfg(target_os = "openbsd")]
+#[derive(Default)]
+pub struct Vec {
+    data: Box<arrayvec::ArrayVec<[u8; 4096]>>
 }
 
 impl Vec {
